libpoolprotocols/stratum/EthStratumClient.cpp

```diff
@@ -624,5 +638,8 @@ void EthStratumClient::submitSolution(Solution solution) {
 			boost::bind(&EthStratumClient::handleResponse, this,
 				boost::asio::placeholders::error));
 	}
+	m_response_pending = true;
+	m_responsetimer.expires_from_now(boost::posix_time::seconds(2));
```

Comment:
while i like the idea of a timeout for responses, i think 2 seconds are to short.

Reply:
It's what BitFly recommends in for all their pools. From a BitFly "Medium" article that I can't seem to find right now. It's what prompted me to include the timeout.